bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -181,6 +181,7 @@ public GarbageCollectorThread(ServerConfiguration conf,
                 }
                 gcStats.getDeletedLedgerCounter().inc();
                 ledgerStorage.deleteLedger(ledgerId);
+                BookieImpl.removeLedgerIdFromMasterKeyCache(ledgerId);
```

Comment:
have to at least move this to a Consumer<Long> and decuple GarbageCollectorThread from BookieImpl

also, having a static variables means that if you run multiple bookies in a test case (here in the BK repository) the values of the cache a messed up. Probably it is not a big deal but it is a big code smell

Reply:
Good suggestion, I have updated